src/plugins/visualizations/server/usage_collector/get_usage_collector.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Observable } from 'rxjs';
+import { countBy, get, groupBy, mapValues, max, min, values } from 'lodash';
+import { first } from 'rxjs/operators';
+import { SearchResponse } from 'elasticsearch';
+
+import { LegacyAPICaller } from 'src/core/server';
+import { getPastDays } from './get_past_days';
+
+const VIS_USAGE_TYPE = 'visualization_types';
+
+type ESResponse = SearchResponse<{ visualization: { visState: string } }>;
+
+interface VisSummary {
+  type: string;
+  space: string;
+  past_days: number;
+}
+
+/*
+ * Parse the response data into telemetry payload
+ */
+async function getStats(callCluster: LegacyAPICaller, index: string) {
+  const searchParams = {
+    size: 10000, // elasticsearch index.max_result_window default value
+    index,
+    ignoreUnavailable: true,
+    filterPath: [
+      'hits.hits._id',
+      'hits.hits._source.visualization',
+      'hits.hits._source.updated_at',
+    ],
+    body: {
+      query: {
+        bool: { filter: { term: { type: 'visualization' } } },
+      },
+    },
+  };
+  const esResponse: ESResponse = await callCluster('search', searchParams);
+  const size = get(esResponse, 'hits.hits.length');
+  if (size < 1) {
```

Comment:
@sulemanof I've noticed this error ever since this PR has been merged (for fresh installed clusters):

```
server    log   [10:30:55.212] [warning][collector-set][plugins][usageCollection] TypeError: Cannot read property 'hits' of undefined
    at getStats (/src/plugins/visualizations/server/usage_collector/get_usage_collector.ts:64:54)
    at process._tickCallback (internal/process/next_tick.js:68:7)
server    log   [10:30:55.216] [warning][collector-set][plugins][usageCollection] Unable to fetch data from visualization_types collector
```

The reason: `size` is `undefined` and `undefined < 1` is `false`.

Reply:
I've created https://github.com/elastic/kibana/pull/77478 to tackle this